### PR TITLE
Chat with api

### PR DESCRIPTION
**Pull request by dnaranjo89**
Use the API to fetch and send chat messages

Return the alias in the entries of the chat (in addition of the user_id, avatar and message)

### CODE REVIEW

**Comment by Mariocj89** [comment]
Dont use update. Use a list comprebhension. [User_map[msg[user]] for msg in messages]

**Comment by Mariocj89** [comment]
Ok. I ser the issue know ... You want to update justo parte of the mesaage (user related ) then use a dict comprehension :)

**Comment by dnaranjo89** [comment]
why using a dict comprehesion if I just want to add a couple of elements to the existing dict?

**Comment by Mariocj89** [comment]
Please chabge name attribute yo user or the like as it is misleading.also reutrn ab empty dict instead of empty string forma consistency.
Moreover, i dont like TVE update method at al for that kind of usecase. It is obscure. You dont know what u update ckearly ( but it is fine if u dont want yo change that)

**Comment by dnaranjo89** [comment]
Do you mean change alias to user_alias?

And how would you do it with a comprehension dict?

File: web/rest_api/draw.py [hunk 1]
```diff
@@ -119,29 +119,32 @@ def chat(self, request, **kwargs):
...
         elif request.method == 'GET':
             # TODO: return only the list of messages
-            def get_user_image(username):
+            def get_user_details(username):
                 """function to get either the user avatar or an empty string"""
                 try:
-                    return self._client.retrieve_user(username).user_image
+                    user = self._client.retrieve_user(username)
                 except Exception:
                     return ""
```
Comment (Mariocj89) [inline web/rest_api/draw.py]: return {} instead of "" for consistency

File: web/rest_api/draw.py [hunk 2]
```diff
@@ -119,29 +119,32 @@ def chat(self, request, **kwargs):
...
+                else:
+                    return {"alias": user.alias, "avatar": user.user_image}
             try:
                 messages = self._client.retrieve_chat_messages(draw_id)
             except mongodb.MongoDriver.NotFoundError:
                 messages = []
 
             users = set([message["user"] for message in messages])
-            users_map = {name: get_user_image(name) for name in users}
+            users_map = {name: get_user_details(name) for name in users}
```
Comment (Mariocj89) [inline web/rest_api/draw.py]: name is going to be a dict with alias an avatar??? isnt that weird?

File: web/rest_api/draw.py [hunk 3]
```diff
@@ -119,29 +119,32 @@ def chat(self, request, **kwargs):
...
+
             for message in messages:
-                message["avatar"] = users_map[message["user"]]
+                message.update(users_map[message["user"]])
```
Comment (Mariocj89) [inline web/rest_api/draw.py]: this is misleading, you are just setting ONE attribute in the dictionary and using a function that does a lot more than that. please change it.

File: web/rest_api/draw.py (hunk 2 above)
Comment (dnaranjo89) [inline web/rest_api/draw.py]: I missed that

File: web/rest_api/draw.py (hunk 3 above)
Comment (dnaranjo89) [inline web/rest_api/draw.py]: I'm adding two elements to the dict (alias and avatar). How else can be done?

Comment (dnaranjo89) [inline web/rest_api/draw.py]: I'm adding two attributes (alias and avatar). How else can be done?

File: web/rest_api/draw.py (hunk 2 above)
Comment (dnaranjo89) [inline web/rest_api/draw.py]: user_map is a dict that has user_ids as key and their details as values

Comment (Mariocj89) [inline web/rest_api/draw.py]: sorry, I did too much js lately >.<

**Pull request merged**